rsciio/trivista/_api.py

```diff
@@ -0,0 +1,745 @@
+            if isinstance(entry_dict_old, list):
+                for item in entry_dict_old:
+                    entry_dict_new[item["Name"]] = item["Value"]
+            elif isinstance(entry_dict_old, dict):
+                entry_dict_new[entry_dict_old["Name"]] = entry_dict_old["Value"]
+            if entry_name in result:
+                save_name_for_rename.add(entry_name)
+                entry_name += str(idx + 1)
+            result[entry_name] = entry_dict_new
+    for name in save_name_for_rename:
+        result[name + "1"] = result.pop(name)
+    return result
+
+
+class TrivistaTVFReader:
+    """Class to read Trivista's .tvf-files.
+
+    The file is read using xml.etree.ElementTree.
+
+    Parameters
+    ----------
+    file_path: pathlib.Path
+        Path to the to be read file.
+
+    Attributes
+    ----------
+    data, metadata, original_metadata, axes
+
+    Methods
+    -------
+    read_file
+    """
+
+    def __init__(self, file_path):
+        self._file_path = file_path
+
+        if importlib.util.find_spec("lumispy") is None:
+            self._lumispy_installed = False
+            _logger.warning(
+                "Cannot find package lumispy, using BaseSignal as signal_type."
+            )
+        else:
+            self._lumispy_installed = True  # pragma: no cover
+
+    @property
+    def _signal_type(self):
+        if self._lumispy_installed:
+            return "Luminescence"  # pragma: no cover
+        else:
+            return ""
+
+    @property
+    def num_datasets(self):
+        return int(self.data_head.findall("Childs")[0].attrib["Count"])
+
+    def parse_file_structure(self, filter_original_metadata):
+        """Initial parse through the file to extract original metadata and the data location.
+        In general the file structure looks something like this.
+        - root_level
+            - root_level metadata
+                - FileInfoSerialized
+            - Document
+                - data/signal_axis
+                - document metadata
+                    - InfoSerialized
+            - Hardware
+
+        Most of the usable metadata is in the InfoSerialized section.
+        InfoSerialized and FileInfoSerialized need to be converted extra
+        with _etree_to_dict().
+
+        The metadata from the hardware section contains information for all
+        available hardware (i.e. multiple objectives even though only one is used.
+        In the filtering process, the metadata of the actual used objective is extracted).
+
+        Parameters
+        ----------
+        filter_original_metadata: bool
+            if True then 2 seperate dicts are created
+            (original_metadata, unfiltered_original_metadata).
+            Otherwise only the former is created.
+            Filtering original_metadata is done independent
+            of the value of this parameter.
+            This ensures independence of the metadata mapping
+            on this setting.
```

Comment:
This is not the behavior that was retained in the end I think. Consider updating